pkg/controller/machine/machine_controller_test.go

```diff
@@ -17,87 +17,243 @@ limitations under the License.
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      "foo",
-			Namespace: "default",
-			Labels: map[string]string{
-				machinev1.MachineClusterIDLabel: "foo",
+	BeforeEach(func() {
+		By("Setting up a new manager")
+		mgr, err := manager.New(cfg, manager.Options{
+			Metrics: server.Options{
+				BindAddress: "0",
 			},
-		},
-		Spec: machinev1.MachineSpec{
-			ProviderSpec: machinev1.ProviderSpec{
-				Value: &runtime.RawExtension{
-					Raw: []byte("{}"),
+		})
+		Expect(err).NotTo(HaveOccurred())
+
+		k8sClient = mgr.GetClient()
+		k = komega.New(k8sClient)
+
+		By("Setting up feature gates")
+		defaultMutableGate := feature.DefaultMutableFeatureGate
+		_, err = features.NewFeatureGateOptions(defaultMutableGate, openshiftfeatures.SelfManaged, openshiftfeatures.FeatureGateMachineAPIMigration)
+		Expect(err).NotTo(HaveOccurred())
+
+		err = defaultMutableGate.SetFromMap(map[string]bool{"MachineAPIMigration": true})
+		Expect(err).NotTo(HaveOccurred())
+
+		By("Setting up a new reconciler")
+		act := newTestActuator()
+		reconciler := newReconciler(mgr, act, defaultMutableGate)
+
+		Expect(add(mgr, reconciler, "testing")).To(Succeed())
+
+		var mgrCtx context.Context
+		mgrCtx, mgrCtxCancel = context.WithCancel(ctx)
+		mgrStopped = make(chan struct{})
+
+		By("Starting the manager")
+		go func() {
+			defer GinkgoRecover()
+			defer close(mgrStopped)
+
+			Expect(mgr.Start(mgrCtx)).To(Succeed())
+		}()
+
+		By("Creating the namespace")
+		namespace = &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{GenerateName: "machine-test"}}
+		Expect(k8sClient.Create(ctx, namespace)).To(Succeed())
+
+		By("Setting up the machine builder")
+		machineBuilder = machinev1resourcebuilder.Machine().
+			WithNamespace(namespace.ObjectMeta.Name).
+			WithGenerateName("foo").
+			WithProviderSpecBuilder(machinev1resourcebuilder.GCPProviderSpec())
+
+	})
+
+	AfterEach(func() {
+		By("Deleting the machines")
+		Expect(cleanResources()).To(Succeed())
+
+		By("Deleting the namespace")
+		Expect(k8sClient.Delete(ctx, namespace)).To(Succeed())
+
+		By("Closing the manager")
+		mgrCtxCancel()
+		Eventually(mgrStopped, timeout).WithTimeout(20 * time.Second).Should(BeClosed())
+	})
+
+	It("Should reconcile a Machine", func() {
+		instance := machineBuilder.Build()
+
+		By("Creating the Machine")
+		Expect(k8sClient.Create(ctx, instance)).To(Succeed())
+
+		By("Setting the AuthoritativeAPI to MachineAPI")
+		Eventually(k.UpdateStatus(instance, func() {
+			instance.Status.AuthoritativeAPI = machinev1.MachineAuthorityMachineAPI
+		})).Should(Succeed())
+
+		Eventually(k.Object(instance), timeout).Should(HaveField("Status", Not(Equal(machinev1.MachineStatus{}))))
+
+		// // Expect platform status is not empty. This (check) means we've called the
+		// // actuator. Given our test actuator doesn't set any status, we may need to expect to be called.
+
+		// Eventually(k.Object(instance), timeout).Should(HaveField("Status.ProviderStatus", Not(Equal(runtime.RawExtension{}))))
```

Comment:
This is todo before merge?

Reply:
I think we should leave this for another time. The current test actuator only counts calls, it doesn't actually do anything like setting a status.

Reply:
Probably want some explanation of that then I guess?